upup/pkg/fi/cloudup/azure/azure_cloud.go

```diff
@@ -128,9 +133,39 @@ func (c *azureCloudImplementation) AddClusterTags(tags map[string]*string) {
 }
 
 func (c *azureCloudImplementation) GetApiIngressStatus(cluster *kops.Cluster) ([]kops.ApiIngressStatus, error) {
-	// TODO(kenji): Implement this. Currently we return nil as we
-	// don't create any resources for ingress to the API server.
-	return nil, nil
+	var ingresses []kops.ApiIngressStatus
+	var rg string = cluster.AzureResourceGroupName()
+
+	// Get scale sets in cluster resource group and find masters scale set
+	scaleSets, err := c.vmscaleSetsClient.List(context.TODO(), rg)
+	if err != nil {
+		return nil, fmt.Errorf("error getting Cluster Master Scale Set for API Ingress Status: %s", err)
+	}
+	var vmssName string
+	for _, scaleSet := range scaleSets {
+		val, ok := scaleSet.Tags["KubernetesCluster"]
```

Comment:
How about 

```go
val, ok := scaleSet.Tags[TagClusterName]
val2, ok2 := scaleSet.Tags[TagNameRolePrefix + TagRoleMaster]
```

Reply:
I like this better too. When looking over recent comments in the remaining azure work issue I started having concern about this.  Thanks for the suggestion 👍